src/argbash-init.m4

```diff
@@ -204,45 +180,41 @@ do_body_protected()
 	echo "# ] <-- needed because of Argbash"
 }
 
-
 do_stuff()
 {
 	do_header "$1"
 	do_args
 	do_args_footer
 
-	test "$_arg_separate" = 0 && do_body_protected
+	test "${_arg_separate}" = 0 && do_body_protected
 }
 
-outfname="$_arg_output"
+outfname="${_arg_output}"
 # we canonicize the empty string input to output filename to the dash
```

Comment:
Spellcheck highlighted canonicize here, so I was going to wordsmith this some, but then I realized I don't quite understand what you're saying.  If you'd like to elaborate I might be able to put something in here that is easier to understand

Reply:
I hope that this answers your question:
canonicize = make canonical, i.e. if the output file is stdout, then the canonical representation of thereof is dash. Therefore, all other representations (currently only an empty string) are converted to canonical one - the dash.

I guess that it could be performed more efficiently by Bash parameter substitution.

Reply:
Ah you meant canonicalize.

So I'm going to change this to say:
`# Alter the output filename to the canonical representation of stdout (-) if it is the empty string`

Reply:
I think that helps.  Also, just for reference <https://stackoverflow.com/a/16753536> - POSIX substitution rules for variables in a simple to lookup table.